lib/Target/SubstraitPB/Export.cpp

```diff
@@ -137,6 +141,115 @@ FailureOr<std::unique_ptr<Rel>> exportOperation(CrossOp op) {
+    auto childReference = std::move(referenceRoot);
+    referenceRoot = std::make_unique<ReferenceSegment>();
+
+    // Create `StructField` message.
+    // TODO(ingomueller): support other segment types.
+    auto indexAttr = llvm::cast<IntegerAttr>(attr);
+    auto structField = std::make_unique<ReferenceSegment::StructField>();
+    structField->set_field(indexAttr.getInt());
+    structField->set_allocated_child(childReference.release());
+
+    referenceRoot->set_allocated_struct_field(structField.release());
+  }
+
+  // Build `FieldReference` message.
+  auto fieldReference = std::make_unique<FieldReference>();
+  fieldReference->set_allocated_direct_reference(referenceRoot.release());
+  // TODO: support other reference types.
+  auto rootReference = std::make_unique<FieldReference::RootReference>();
+  fieldReference->set_allocated_root_reference(rootReference.release());
+
+  // Build `Expression` message.
+  auto expression = std::make_unique<Expression>();
+  expression->set_allocated_selection(fieldReference.release());
+
+  return expression;
+}
+
+FailureOr<std::unique_ptr<Rel>> exportOperation(FilterOp op) {
+  // Build `RelCommon` message.
+  auto relCommon = std::make_unique<RelCommon>();
+  auto direct = std::make_unique<RelCommon::Direct>();
+  relCommon->set_allocated_direct(direct.release());
+
+  // Build input `Rel` message.
+  auto inputOp =
+      llvm::dyn_cast_if_present<RelOpInterface>(op.getInput().getDefiningOp());
+  if (!inputOp)
+    return op->emitOpError("input was not produced by Substrait relation op");
+
+  FailureOr<std::unique_ptr<Rel>> inputRel = exportOperation(inputOp);
+  if (failed(inputRel))
+    return failure();
+
+  // Build condition `Expression` message.
+  auto yieldOp = llvm::cast<YieldOp>(op.getCondition().front().getTerminator());
+  // TODO(ingomueller): There can be cases where there isn't a defining op but
+  //                    the region argument is returned directly. Support that.
+  auto conditionOp = llvm::dyn_cast_if_present<ExpressionOpInterface>(
+      yieldOp.getValue().getDefiningOp());
+  if (!conditionOp)
+    return op->emitOpError("condition not supported for export: yielded op was "
+                           "not produced by Substrait expression op");
+  FailureOr<std::unique_ptr<Expression>> condition =
+      exportOperation(conditionOp);
+  if (failed(condition))
+    return failure();
+
+  // Build `FilterRel` message.
+  auto filterRel = std::make_unique<FilterRel>();
+  filterRel->set_allocated_common(relCommon.release());
+  filterRel->set_allocated_input(inputRel->release());
+  filterRel->set_allocated_condition(condition->release());
+
+  // Build `Rel` message.
+  auto rel = std::make_unique<Rel>();
+  rel->set_allocated_filter(filterRel.release());
+
+  return rel;
+}
+
+FailureOr<std::unique_ptr<Expression>> exportOperation(LiteralOp op) {
+  auto si1 = IntegerType::get(op.getContext(), 1, IntegerType::Signed);
```

Comment:
Given you are just using for checking, you could do a dyn\_cast\<IntegerType> and check signed and type (these 2 here would be cached, but probably cheaper check still)

Reply:
Good point! This is part of #812, so I addressed it there.